device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -215,6 +216,24 @@ public DeviceClient(String connString, IotHubClientProtocol protocol, String pub
         commonConstructorSetup();
     }
 
+    /**
+     * Creates a device client that uses the provided SSLContext for SSL negotiation
+     * @param connString the connection string for the device. May be an x509 connection string (format: "HostName=...;DeviceId=...;x509=true")
```

Comment:
Why do we need this change? Adding a new set option complicates our testing. Why do we need our customers to be able setup this API?

Reply:
Originally, I was going to make this exposed via SetOption like we do with the other certificate settings, but now I just have a constructor that takes the SSLContext. The description of this PR was out of date when you read that it was going to be a new SetOption option.

This new constructor is important for our customers because it allows them to control certificate parsing. Things like ECC certs that we don't have logic for can be done by customer instead. Also, there are security concerns with asking our users to provide certificates in file or string form as mentioned in issue #578